mp/src/game/client/momentum/c_mom_player.cpp

```diff
@@ -122,6 +131,32 @@ void C_MomentumPlayer::PostDataUpdate(DataUpdateType_t updateType)
     BaseClass::PostDataUpdate(updateType);
 }
 
+void C_MomentumPlayer::SetRampBoardVelocity(Vector vecVel)
+{
+    m_vecRampBoardVel = vecVel;
+    m_bSurfing = true;
+
+    IGameEvent *pEvent = gameeventmanager->CreateEvent("ramp_board");
+    if (pEvent)
+    {
+        pEvent->SetFloat("speed", vecVel.Length());
+        gameeventmanager->FireEvent(pEvent);
+    }
+}
```

Comment:
Could this move to mom_player_shared ? See `FireBullet`

Reply:
Also, does this event need to be fired on both server and client side?

Reply:
Moved this to `mom_player_shared` as yeah it does the same stuff. Put an `ifdef` around the event firing as only the server needs to fire that yeah.